src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -74,6 +73,10 @@ internal ConversionsBase WithNullability(bool includeNullability)
 
         protected abstract Conversion GetInterpolatedStringConversion(BoundUnconvertedInterpolatedString source, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo);
 
+#nullable  enable
+        public abstract bool IsApplicableInterpolatedStringBuilderType(BoundUnconvertedInterpolatedString source, TypeSymbol builderType, BindingDiagnosticBag diagnostics, [NotNullWhen(true)] out ImmutableArray<MethodArgumentInfo> builderArguments);
```

Comment:
>IsApplicableInterpolatedStringBuilderType [](start = 29, length = 41)

It looks like this method is not used to classify conversion, but rather performs some part of binding work and is dependent on a Binder. I think we should move it to the Binder.

Reply:
> It looks like this method is not used to classify conversion, but rather performs some part of binding work and is dependent on a Binder. I think we should move it to the Binder.

The reason I have it here because I believe I am going to need it when doing overload resolution, when determining whether an interpolated string expression is convertible to a parameter type.

---
In reply to: [610785056](https://github.com/dotnet/roslyn/pull/52422#discussion_r610785056) [](ancestors = 610785056)

Reply:
>> It looks like this method is not used to classify conversion, but rather performs some part of binding work and is dependent on a Binder. I think we should move it to the Binder.

> The reason I have it here because I believe I am going to need it when doing overload resolution, when determining whether an interpolated string expression is convertible to a parameter type.

I think we should not add the API until then. Also, given what the method is doing, I think it belongs in the Binder.

---
In reply to: [612032667](https://github.com/dotnet/roslyn/pull/52422#discussion_r612032667) [](ancestors = 612032667,610785056)